sentry/src/main/java/io/sentry/JsonObjectReader.java

```diff
@@ -0,0 +1,56 @@
+package io.sentry;
+
+import io.sentry.vendor.gson.stream.JsonReader;
+import io.sentry.vendor.gson.stream.JsonToken;
+import java.io.IOException;
+import java.io.Reader;
+import java.util.Map;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.Nullable;
+
+@ApiStatus.Internal
+public final class JsonObjectReader extends JsonReader {
+
+  public JsonObjectReader(Reader in) {
+    super(in);
+  }
+
+  public @Nullable String nextStringOrNull() throws IOException {
+    return peek() == JsonToken.NULL ? null : nextString();
+  }
+
+  public @Nullable Double nextDoubleOrNull() throws IOException {
+    return peek() == JsonToken.NULL ? null : nextDouble();
+  }
+
+  public @Nullable Long nextLongOrNull() throws IOException {
+    return peek() == JsonToken.NULL ? null : nextLong();
+  }
+
+  public @Nullable Integer nextIntegerOrNull() throws IOException {
+    return peek() == JsonToken.NULL ? null : nextInt();
+  }
+
+  public @Nullable Boolean nextBooleanOrNull() throws IOException {
+    return peek() == JsonToken.NULL ? null : nextBoolean();
+  }
+
+  public void nextUnknown(ILogger logger, Map<String, Object> unknown, String name) {
+    try {
+      unknown.put(name, nextObjectOrNull());
+    } catch (Exception exception) {
+      String message = "Error deserializing unknown key \"" + name + "\"";
+      logger.log(SentryLevel.ERROR, message, exception);
```

Comment:
iirc `.log` already has an argument that allows you to use something similar to String.format, eg `"message: %s", message`